gems/pending/spec/disk/modules/azure_blob_disk_spec.rb

```diff
@@ -0,0 +1,152 @@
+require 'ostruct'
+require 'azure-armrest'
+require 'disk/modules/AzureBlobDisk'
+require 'vcr'
+
+describe AzureBlobDisk do
+  before(:all) do
+    @test_env = TestEnvHelper.new(__FILE__)
+    @test_env.vcr_filter
+
+    @client_id     = @test_env[:azure_client_id]
+    @client_key    = @test_env[:azure_client_key]
+    @tenant_id     = @test_env[:azure_tenant_id]
+    @blob_uri      = @test_env[:blob_uri]
+
+    @test_env.ensure_recording_dir_exists
+  end
+
+  before(:each) do |example|
+    Azure::Armrest::ArmrestService.clear_caches
+    example_id = "#{example.example_group.description}-#{example.metadata[:ex_tag]}"
+    cassette_name = @test_env.cassette_for(example_id)
+    VCR.insert_cassette(cassette_name, :decode_compressed_response => true)
+
+    @azure_config = Azure::Armrest::ArmrestService.configure(
+      :client_id  => @client_id,
+      :client_key => @client_key,
+      :tenant_id  => @tenant_id
+    )
+
+    @storage_account_service = Azure::Armrest::StorageAccountService.new(@azure_config)
+  end
+
+  after(:each) do
+    VCR.eject_cassette
+  end
+
+  describe ".new" do
+    it "should raise an error, given a bad blob uri", :ex_tag => 1 do
```

Comment:
Right.  `cassette_for` is really the consumer of this value + the example group description and this is not obvious.  Maybe if I say `example_group_cassette_suffix`, you'll come up with a much better name for the tag. :wink: RSpec tags can also be used to run just examples with that tag so keep in mind that if we name it right, it will double as documentation of how to name your cassettes but also how to invoke just those tests.